### PR TITLE
add Python 3.9 support

### PR DESCRIPTION
**Pull request by graingert**
### Contributor Checklist:

- [ ] I have updated the release notes at `docs/source/NEWS.rst`
- [x] I have updated the automated tests.
- [x] All tests pass on your local dev environment. See `CONTRIBUTING.rst`.

### CODE REVIEW

**Comment by graingert** [comment]
currently twisted latest doesn't support py3.9 on windows, need to wait for a new twisted release

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -44,6 +44,11 @@ jobs:
         with:
           python-version: 3.8
 
+      - name: Set Up Python 3.9
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9.0-rc.2
```
Comment (graingert) [inline .github/workflows/main.yml:50]:
```suggestion
          python-version: 3.9
```

File: .readthedocs.yml [hunk 2]
```diff
@@ -7,8 +7,11 @@ formats:
   - pdf
   - epub
 
+build:
+  image: testing
+
```
Comment (graingert) [inline .readthedocs.yml:12]:
```suggestion
```

File: .readthedocs.yml [hunk 3]
```diff
@@ -7,8 +7,11 @@ formats:
...
 python:
-  version: 3.8
+  version: 3.9
```
Comment (graingert) [inline .readthedocs.yml:14]:
```suggestion
  version: 3.8
```

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -78,7 +83,7 @@ jobs:
           TOX_PARALLEL_NO_SPINNER: 1
         run:
           tox --parallel=all
-          -e'py{35,36,37,38,py3}-test-{twlatest,twtrunk},publishcov,py38-linters,release'
+          -e'py{35,36,37,38,39,py3}-test-{twlatest,twtrunk},publishcov,py39-linters,release'
```
Comment (graingert) [inline .github/workflows/main.yml:86]:
```suggestion
          -e'py39-test-twtrunk,py{35,36,37,38,py3}-test-{twlatest,twtrunk},publishcov,py39-linters,release'
```

**Comment by adiroiban** [comment]
Do you know why Twisted doesn't support 3.9 . Is there a ticket for twisted/twisted to add support for 3.9?

Thanks

**Comment by graingert** [comment]
> Do you know why Twisted doesn't support 3.9 . Is there a ticket for twisted/twisted to add support for 3.9?
> 
> Thanks

twlatest doesn't support 3.9, but twtrunk does

**Comment by adiroiban** [comment]
> twlatest doesn't support 3.9, but twtrunk does

ah... thanks. well. I hope we will do a twisted release soon :)

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -76,7 +81,7 @@ jobs:
           TOX_SKIP_ENV: ${{ matrix.cfg.tox-skip-env }}
         run:
           tox --parallel=all
-          -e'py{35,36,37,38,py3}-test-{twlatest,twtrunk},publishcov,py38-linters,release'
+          -e'py39-test-twtrunk,py{35,36,37,38,py3}-test-{twlatest,twtrunk},publishcov,py39-linters,release'
```
Comment (graingert) [inline .github/workflows/main.yml:84]:
```suggestion
          -e'py{35,36,37,38,39,py3}-test-{twlatest,twtrunk},publishcov,py39-linters,release'
```

File: pyproject.toml [hunk 6]
```diff
@@ -106,5 +106,5 @@ deps =
   coverage[toml]>=5.3
 commands =
     codecov --required
-depends = py{,38,34,35,36,37,py3}-test-{twlatest,twtrunk,dev}
+depends = py{,35,36,37,38,39,py3}-test-{twlatest,twtrunk,dev}
```
Comment (adiroiban) [inline pyproject.toml:109]: is there any semantic for the leading comma ? Just tasking :)

**Review by adiroiban — APPROVED** [review]
All good. Thanks! I guess that py3.5 removal will follow soon.

File: pyproject.toml (hunk 6 above)
Comment (graingert) [reply pyproject.toml:109]: ah this is for native python eg `py-test-twlatest`

**Pull request merged**